File: week_2_Dizainier/Dizainier/Dizainier/ViewController.m
//
//  ViewController.m
//  Dizainier
//
//  Created by Nicolas Salleron on 25/09/2017.
//  Copyright © 2017 Nicolas Salleron. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    [_stepper setValue:0];
    [_Dizaines setSelectedSegmentIndex:([_stepper value]/10)];
    [_Unites setSelectedSegmentIndex:fmod([_stepper value], 10)];
    
    NSLog(@"Dizaine : %ld et Unités : %ld",(long)[_Dizaines selectedSegmentIndex],(long)[_Unites selectedSegmentIndex]);
    
    NSLog(@"Description Dizaine : %@",[_Dizaines description]);
    NSLog(@"Description Unites : %@",[_Unites description]);
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)changeMode:(id)sender {
    [self affichageResult];
}
- (IBAction)fctCalc:(UISegmentedControl *)sender {
    //NSLog(@"=> Description %@",[sender description]);
    
    
    if([sender isEqual:_Dizaines]){
        
        
        NSLog(@"Dizaines %ld", (int)fmod([_stepper value], 10) +(int)10*[sender selectedSegmentIndex]);
        [_stepper setValue:(int)fmod([_stepper value], 10)
         +(int)10*[sender selectedSegmentIndex]];
        
        
    }else if([sender isEqual:_Unites]){
        NSLog(@"Stepper value : %d Unites %ld", (int) [_stepper value], [sender selectedSegmentIndex]);
        [_stepper setValue:(int)(((int)[_stepper value])/10)*10
                            +[sender selectedSegmentIndex]];
        
        NSLog(@"Result : %f",[_stepper value]);
        
    }else{  //Demande d'actualisation d'affichage;
        [_Dizaines setSelectedSegmentIndex:[_stepper value]/10];
        [_Unites setSelectedSegmentIndex:fmod([_stepper value], 10)];
        [_Slider setValue:[_stepper value]];
    }
   
    
    [self affichageResult];
    
    
}
- (IBAction)changeStepper:(id)sender {
    [self fctCalc:nil];
    NSLog(@"Valeur Stepper : %d",(int)[_stepper value]);
}

- (IBAction)changeSlider:(UISlider*)sender {
  
}
- (IBAction)btnRAZ:(UIButton *)sender {
    [_stepper setValue:0];
    [self fctCalc:nil];
}

-(void) affichageResult{
    int result = [_stepper value];
    
    //Affichage du resultat
    if([_modeGeek isOn]){
        [_result setText:[NSString stringWithFormat:@"%x",result]];
    }else{
        [_result setText:[NSString stringWithFormat:@"%d",result]];
    }
    
    //Affichage de la couleur
    if( result == 42)
        [_result setTextColor:[UIColor redColor]];
    else
        [_result setTextColor:[UIColor blackColor]];
    
    [_Slider setValue:result];
    
}

- (void)dealloc {
    [_Dizaines release];
    [_Unites release];
    [_stepper release];
    [_modeGeek release];
    [_Slider release];
    [_btnRAZ release];
    [_result release];
    [_Unites release];
    [super dealloc];
}
@end

File: week_3_Dizainier/Dizainier/Dizainier/ViewController.swift
//
//  ViewController.swift
//  Dizainier
//
//  Created by Nicolas Salleron on 01/10/2017.
//  Copyright © 2017 Nicolas Salleron. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var scDizaines: UISegmentedControl!
    @IBOutlet weak var btnStepper: UIStepper!
    @IBOutlet weak var scUnites: UISegmentedControl!
    @IBOutlet weak var btnSwitch: UISwitch!
    @IBOutlet weak var sdSlider: UISlider!
    @IBOutlet weak var labelResult: UILabel!
    
    @IBOutlet weak var btnraz: UIButton!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        labelResult.text = "";
        btnStepper.value = 0;
        btnSwitch.isOn = false;
        
        scDizaines.selectedSegmentIndex = Int(btnStepper.value/10.0);
        scUnites.selectedSegmentIndex = Int(fmod(btnStepper.value, 10.0));
    
        
    }

    @IBAction func actionSW(_ sender: Any) {
        self.affichageResult()
    }
    
    
    
    @IBAction func btnRAZ(_ sender: Any) {
        btnStepper.value = 0;
        sdSlider.value = 0
        scUnites.selectedSegmentIndex = 0
        scDizaines.selectedSegmentIndex = 0
        self.affichageResult()
    }
    
    @IBAction func actionStepper(_ sender: UIStepper) {
        self.calcul(nil);
    }
    
    
    @IBAction func actionSC(_ sender: UISegmentedControl) {
        self.calcul(sender)
    }
    
    @IBAction func actionSlider(_ sender: Any) {
        btnStepper.value = Double(Int(sdSlider.value))
        self .calcul(nil)
        
        
    }
    
    func calcul(_ sender: UISegmentedControl?){
        
        if( sender === scDizaines){
            let unites = Int(fmod(btnStepper.value, 10))
            let dizaine = Int((sender?.selectedSegmentIndex)! * 10)
            btnStepper.value = Double(dizaine+unites)
        }else if (sender === scUnites) {

            let dizaine = Int((btnStepper.value/10))
            let unites = Int((sender?.selectedSegmentIndex)!)
            btnStepper.value = Double(Double(dizaine)*10+Double(unites))
            
        }else{
            scDizaines.selectedSegmentIndex = Int(btnStepper.value/10)
            scUnites.selectedSegmentIndex = Int(fmod(btnStepper.value, 10.0))
            btnStepper.value = Double(scDizaines.selectedSegmentIndex*10 + scUnites.selectedSegmentIndex);
            sdSlider.setValue(Float(btnStepper.value), animated: true)
            
        }
        
        self.affichageResult()
        
    }
    
    
    func affichageResult () {
        let result = btnStepper.value
        
        if (btnSwitch.isOn){
            let hex = String(format:"%02X",Int(result))
            labelResult.text = hex
        }else{
            labelResult.text = "\(Int(result))"
        }
        
        if(result == 42){
            labelResult.textColor = UIColor.red
        }else{
            labelResult.textColor = UIColor.black
        }
        
        sdSlider.setValue(Float(result), animated: true)
        
      
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

